src/renderer.rs

```diff
@@ -320,4 +325,71 @@ impl Renderer {
             line(color::BLACK.into(), 1., [start[0], start[1], end[0], end[1]], c.transform, g);
         }
     }
+
+    /// Save the drawings to SVG
+    fn save_svg<P: AsRef<path::Path>>(&mut self, path: P) {
+        let drawing_state = self.app.drawing();
+        let mut document = svg::Document::new()
+            .set("viewBox", (0, 0, drawing_state.width, drawing_state.height));
+
+        // set background color - https://stackoverflow.com/a/11293812/9276882
+        let bg_color = drawing_state.background;
+        let background = SvgRect::new()
+            .set("width", "100%")
+            .set("height", "100%")
+            .set("fill", format!("rgba({},{},{},{})", bg_color.red, bg_color.green, bg_color.blue, bg_color.alpha));
+
+        document = document.add(background);
+
+        let center = Point {x: drawing_state.width as f64 * 0.5, y: drawing_state.height as f64 * 0.5};
+
+        for drawing in &self.drawings {
+            match *drawing {
+                Drawing::Path(ref path) => {
+                    let &Path { start, end, ref pen } = path;
+                    let &Pen { thickness, color, enabled: _ } = pen;
+
+                    let start = start.to_screen_coords(center);
+                    let end = end.to_screen_coords(center);
+
+                    let line = SvgLine::new()
+                        .set("x1", start.x)
+                        .set("y1", start.y)
+                        .set("x2", end.x)
+                        .set("y2", end.y)
+                        .set("stroke", format!("rgba({},{},{},{})", color.red, color.green, color.blue, color.alpha))
+                        .set("stroke-width", format!("{}px", thickness * 2.0));
+
+                    // Note that in above code `stroke-width` is twice the `thickness`.
+                    // As per https://docs.piston.rs/piston_window/piston_window/fn.line.html,
+                    // second param of piston_window::line is radius.
+                    // So radius (thickness) = 1 takes 2 pixels on screen.
+                    // Hence, stroke-width of svg element is double the thickness.
+
+                    document = document.add(line);
+                },
+                Drawing::Polygon(ref poly) => {
+                    let points = poly.vertices.iter()
+                        .map(|p| {
+                            let point = p.to_screen_coords(center);
+                            format!("{},{} ", point.x, point.y)
+                        })
+                        .fold("".to_string(), |acc, x| acc + &x);
+
+                    let color = poly.fill_color;
```

Comment:
I think if you add the inner function I mentioned in another comment you may be able to get rid of a lot of the temporary variables that you've been creating in order to format the colors.